### PR TITLE
fix hdrs CommonTools

### PR DESCRIPTION
**Pull request by mrodozov**
Fix headers related to #15248

### CODE REVIEW

**Comment by mrodozov** [comment]
please test

File: CommonTools/PileupAlgos/plugins/PuppiPhoton.h [hunk 1]
```diff
@@ -13,6 +13,8 @@
 #include "FWCore/ParameterSet/interface/ParameterSet.h"
 #include "DataFormats/Math/interface/PtEtaPhiMass.h"
 #include "DataFormats/Candidate/interface/Candidate.h"
+#include "DataFormats/ParticleFlowCandidate/interface/PFCandidate.h"
```
Comment (perrotta) [inline CommonTools/PileupAlgos/plugins/PuppiPhoton.h]: These two includes can be removed from PuppiPhoton.cc, then

File: CommonTools/PileupAlgos/plugins/PuppiProducer.h [hunk 2]
```diff
@@ -12,7 +12,10 @@
 #include "FWCore/Framework/interface/EventSetup.h"
 #include "FWCore/ParameterSet/interface/ParameterSet.h"
 #include "DataFormats/Math/interface/PtEtaPhiMass.h"
-
+#include "DataFormats/Math/interface/LorentzVector.h"
+#include "DataFormats/VertexReco/interface/VertexFwd.h"
```
Comment (perrotta) [inline CommonTools/PileupAlgos/plugins/PuppiProducer.h]: These three includes can be removed from PuppiProducer.cc, then

Comment (mrodozov) [reply CommonTools/PileupAlgos/plugins/PuppiProducer.h]: Ok I'll check more careful what could be removed then

**Comment by mrodozov** [comment]
please test

**Comment by perrotta** [comment]
+1
- Technical
- Jenkins tests pass

**Comment by fabiocos** [comment]
+1

**Comment by fabiocos** [comment]
merge

**Pull request merged**